addons/hr_evaluation/models/hr_evaluation.py

```diff
@@ -0,0 +1,370 @@
+    interview_deadline = fields.Date(string="Final Interview", index=True, track_visibility='onchange')
+    date_close = fields.Datetime(string='Appraisal Deadline', index=True, required=True)
+    appraisal_manager_survey_id = fields.Many2one('survey.survey', string='Manager Appraisal', required=False)
+    appraisal_colleagues_survey_id = fields.Many2one('survey.survey', string="Employee's Appraisal")
+    appraisal_self_survey_id = fields.Many2one('survey.survey', string='Self Appraisal')
+    appraisal_subordinates_survey_id = fields.Many2one('survey.survey', string="collaborate's Appraisal")
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.user.company_id)
+
+    @api.onchange('employee_id')
+    def onchange_employee_id(self):
+        if self.employee_id:
+            self.department_id = self.employee_id.department_id
+            self.appraisal_manager = self.employee_id.appraisal_manager
+            self.appraisal_manager_ids = self.employee_id.appraisal_manager_ids
+            self.appraisal_manager_survey_id = self.employee_id.appraisal_manager_survey_id
+            self.appraisal_colleagues = self.employee_id.appraisal_colleagues
+            self.appraisal_colleagues_ids = self.employee_id.appraisal_colleagues_ids
+            self.appraisal_colleagues_survey_id = self.employee_id.appraisal_colleagues_survey_id
+            self.appraisal_self = self.employee_id.appraisal_self
+            self.appraisal_self_survey_id = self.employee_id.appraisal_self_survey_id
+            self.appraisal_subordinates = self.employee_id.appraisal_subordinates
+            self.appraisal_subordinates_ids = self.employee_id.appraisal_subordinates_ids
+            self.appraisal_subordinates_survey_id = self.employee_id.appraisal_subordinates_survey_id
+
+    @api.one
+    @api.constrains('employee_id', 'department_id', 'date_close')
+    def _check_employee_appraisal_duplication(self):
+        """ Avoid duplication"""
+        if self.employee_id and self.department_id and self.date_close:
+            date_closed = fields.Datetime.from_string(self.date_close)
+            start_datetime = date_closed.replace(day=1)
+            end_datetime = date_closed.replace(day=calendar.monthrange(date_closed.year, date_closed.month)[1])
+            appraisals = self.search([
+                ('employee_id', '=', self.employee_id.id), ('department_id', '=', self.department_id.id),
+                ('date_close', '<=', fields.Datetime.to_string(end_datetime)),
+                ('date_close', '>=', fields.Datetime.to_string(start_datetime))])
+            if len(appraisals) > 1:
+                raise UserError(_("You cannot create more than one appraisal for same Month & Year"))
+
+    @api.multi
+    def subscribe_employee(self):
+        for appraisal in self:
+            employees = [emp.address_home_id.id for emp in appraisal.appraisal_manager_ids if emp.address_home_id]
```

Comment:
i think it's return employee home address ids, name should be like employee_partner_ids